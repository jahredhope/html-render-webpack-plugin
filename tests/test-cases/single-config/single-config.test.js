const MemoryFS = require("memory-fs");
const webpack = require("webpack");
const path = require("path");

const config = require("./webpack.config");
const getDirContentsSync = require("../../utils/getDirContentsSync");

describe("Single webpack build", () => {
  const renderDirectory = path.join(process.cwd(), "dist", "render");

  it("should render a HTML file", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    compiler.run(error => {
      expect(error).toBe(null);
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();
      done();
    });
  });
});
