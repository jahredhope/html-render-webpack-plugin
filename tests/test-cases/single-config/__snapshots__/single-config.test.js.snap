// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single webpack build should render a HTML file 1`] = `
Object {
  "index.html": "<html>
<body>
  Rendered with:&nbsp;
  <code>[
    {
      \\"route\\": \\"\\"
    }
  ]</code>
</body>
</html>",
}
`;
